src/Admin/SiteHealth.php

```diff
@@ -171,6 +207,111 @@ public function persistent_object_cache() {
 		];
 	}
 
+	/**
+	 * Return list of available object cache mechanism.
+	 *
+	 * @return array
+	 */
+	public function check_available_cache() {
+		return [
+			'redis'     => $this->is_site_has_redis(),
+			'memcached' => $this->is_site_has_memcached(),
+			'apcu'      => $this->is_site_has_apcu(),
+		];
+	}
+
+	/**
+	 * Check if site has redis cache mechanism.
+	 *
+	 * @return bool True if redis cache is available otherwise False.
+	 */
+	public function is_site_has_redis() {
+
+		if ( ! class_exists( 'Redis' ) ) {
+			return false;
+		}
+
+		/*
+		 * Try to connect redis server.
+		 */
+		$redis_server = [
+			'host'           => isset( $_SERVER['CACHE_HOST'] ) ? sanitize_text_field( $_SERVER['CACHE_HOST'] ) : '127.0.0.1',
+			'port'           => isset( $_SERVER['CACHE_PORT'] ) ? intval( $_SERVER['CACHE_PORT'] ) : 6379,
+			'auth'           => isset( $_SERVER['CACHE_PASSWORD'] ) ? sanitize_text_field( $_SERVER['CACHE_PASSWORD'] ) : null,
+			'database'       => isset( $_SERVER['CACHE_DB'] ) ? $_SERVER['CACHE_DB'] : 0, // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+			'timeout'        => 1, // Time in seconds.
+			'retry_interval' => 100,
+		];
+
+		// Unix socket connection.
+		if ( file_exists( $redis_server['host'] ) && 'socket' === filetype( $redis_server['host'] ) ) {
+			// Port must be null or socket won't connect.
+			$redis_server['port'] = null;
+		}
+
+		try {
+			$redis = new \Redis();
+
+			$redis->connect(
+				$redis_server['host'],
+				$redis_server['port'],
+				$redis_server['timeout'],
+				null,
+				$redis_server['retry_interval']
+			);
+
+			$response = $redis->ping();
+
+		} catch ( Exception $exception ) {
+			$response = false;
+		}
+
+		return ( ! empty( $response ) );
+	}
+
+	/**
+	 * Check if site has memcache cache mechanism.
+	 *
+	 * @return bool True if memcache cache is available otherwise False.
+	 */
+	public function is_site_has_memcached() {
+
+		if ( ! ( class_exists( 'Memcache' ) || class_exists( 'Memcached' ) ) ) {
+			return false;
+		}
+
+		$has_object_cache = false;
```

Comment:
Similar to my comment above, I don't think it our place to attempt to connect to an external service. Checking if the `Memcache` or `Memcached` class exists seems to be a good enough heuristic.